src/main/scala/engine/SimulationHistory.scala

```diff
@@ -12,11 +14,22 @@ object SimulationHistory{
 
   var history:History = List()
 
-  /**Initiliaze the [[History]] of this simulation
+  var mutations: Option[List[Mutation]] = None
+
+  /** Initialize the [[History]] of this simulation
    * @param environment the initial environment of the first [[Generation]] */
   def initialize(environment: Environment): Unit =
     history = Generation(environment, generateInitialCouple) :: history
 
+  /** Introduce a new mutation */
+  def introduceMutation(mutation: Mutation): Unit = {
+    mutations match {
+      case None => mutations = Some(List(mutation))
+      case _ => mutations = Some(mutation :: mutations.get)
+    }
+    GenesUtils.setAlleleDominance(mutation.geneKind.mutated)
```

Comment:
Non è detto che l'Allele scelto come Dominante sia quello mutato, potrebbe essere anche quello base! Per questo in `Mutation` ci serve il campo booleano `isDominant` che ci dice se la mutazione inserita è dominante o meno

Reply:
Giusto! Ho corretto con un if-else